src/main/java/com/intellij/internal/statistic/eventLog/validator/rules/beans/EventGroupRules.java

```diff
@@ -91,7 +91,9 @@ public Object validateEventData(@NotNull String key,
       for (Map.Entry<?, ?> entry : ((Map<?, ?>)data).entrySet()) {
         Object entryKey = entry.getKey();
         if (entryKey instanceof String) {
-          validatedData.put(entryKey, validateEventData(key + "." + entryKey, entry.getValue(), context));
+          Object validateEventData = validateEventData(key + "." + entryKey, entry.getValue(), context);
```

Comment:
Why don't you change `validateEventData` to handle both field name and value at the same time and return, for instance, a Pair? Detecting if event data field was replaced with validation.undefined_rule doesn't look stable